idiss-csharp/IdissLib/Idiss.cs

```diff
@@ -202,25 +247,46 @@ public static IdentityCreationV1 CreateIdentityObjectV1(Versioned<IpInfo> ipInfo
-            ulong nowTimestamp = Convert.ToUInt64(nowTimestampLong);
-            ulong proofTimestamp = request.idRecoveryRequest.value.timestamp;
-            if (proofTimestamp < nowTimestamp - TimestampDelta || proofTimestamp > nowTimestamp + TimestampDelta)
+            var resultPtr = IntPtr.Zero;
+            var outLength = 0;
+            var outCapacity = 0;
+            try
             {
-                throw new RequestValidationException("Invalid timestamp.");
+                long nowTimestampLong = now.ToUnixTimeSeconds();
+                ulong nowTimestamp = Convert.ToUInt64(nowTimestampLong);
+                ulong proofTimestamp = request.idRecoveryRequest.value.timestamp;
+                if (proofTimestamp < nowTimestamp - TimestampDelta || proofTimestamp > nowTimestamp + TimestampDelta)
+                {
+                    throw new RequestValidationException("Invalid timestamp.");
+                }
+
+                byte[] globalBytes = JsonSerializer.SerializeToUtf8Bytes(global);
+                byte[] requestBytes = JsonSerializer.SerializeToUtf8Bytes(request);
+                byte[] ipInfoBytes = JsonSerializer.SerializeToUtf8Bytes(ipInfo);
+                resultPtr = validate_recovery_request_cs(globalBytes, globalBytes.Length, ipInfoBytes,
+                    ipInfoBytes.Length, requestBytes, requestBytes.Length, out outLength, out outCapacity);
+                if (resultPtr != IntPtr.Zero)
+                {
+                    byte[] outBytes = new byte[outLength];
+                    Marshal.Copy(resultPtr, outBytes, 0, outLength);
+                    var errorString = Encoding.UTF8.GetString(outBytes);
+                    throw new RequestValidationException(errorString);
+                }
             }
-            byte[] globalBytes = JsonSerializer.SerializeToUtf8Bytes(global);
-            byte[] requestBytes = JsonSerializer.SerializeToUtf8Bytes(request);
-            byte[] ipInfoBytes = JsonSerializer.SerializeToUtf8Bytes(ipInfo);
-            int outLength = 0;
-            var resultPtr = validate_recovery_request_cs(globalBytes, globalBytes.Length, ipInfoBytes, ipInfoBytes.Length, requestBytes, requestBytes.Length, out outLength);
-            if (resultPtr != IntPtr.Zero)
+            finally
             {
-                byte[] outBytes = new byte[outLength];
-                Marshal.Copy(resultPtr, outBytes, 0, outLength);
-                var errorString = Encoding.UTF8.GetString(outBytes);
-                throw new RequestValidationException(errorString);
+                FreeNonZeroPtr(resultPtr, outLength, outCapacity);
             }
         }
+        
+        private static void FreeNonZeroPtr(IntPtr outputPtr, int outLength, int outCapacity)
+        {
+            if (outputPtr == IntPtr.Zero)
+            {
+                return;
+            }
+
+            free_array_len_cap(outputPtr, outLength, outCapacity);
```

Comment:
It's a bit odd that here `outLength` and `outCapacity` are also `int` but the method takes `u64` in Rust. Whereas other functions that are imported also use `int` in C# but in Rust take i32.

Reply:
My point was more about the fact that int seems  to be 32-bit and the size of the value should match on both sides of the FFI boundary. Otherwise we might have problems, e.g., it could be that in some platforms a 64-bit value is passed as two 32-bit values in registers.

The signed/unsigned distinction I don't think is a problem since all platforms just use two's complement representation.

But I see you changed the type of the C# function to take a ulong so that resolves this issue.